### PR TITLE
[refactor] Remove decimal v1 related code from code base

### PR DESCRIPTION
**Pull request by yangzhg**
## Proposed changes

remove ALL  DECIMAL V1 type code ， this is a part of #6073 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (#6073) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exprs/expr_context.cpp [hunk 1]
```diff
@@ -202,25 +202,6 @@ void ExprContext::get_value(TupleRow* row, bool as_ascii, TColumnValue* col_val)
         col_val->__set_double_val(*reinterpret_cast<double*>(value));
         break;
 #if 0
```
Comment (morningman) [inline be/src/exprs/expr_context.cpp:204]: # if 0 can be removed?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/ColumnType.java [hunk 2]
```diff
@@ -115,7 +114,8 @@ public static void write(DataOutput out, Type type) throws IOException {
         Preconditions.checkArgument(type.isScalarType(), "only support scalar type serialization");
         ScalarType scalarType = (ScalarType) type;
         if (scalarType.getPrimitiveType() == PrimitiveType.DECIMALV2) {
-            Text.writeString(out, PrimitiveType.DECIMAL.name());
+            // todo rename decimalV2 to decimal. then use PrimitiveType.DECIMALV2.name()
+            Text.writeString(out, "DECIMAL");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/ColumnType.java:118]: Here we can just write DECIMALV2. And in read method, check if it is DECIMAL or DECIMALV2

File: gensrc/thrift/Types.thrift [hunk 3]
```diff
@@ -66,7 +66,7 @@ enum TPrimitiveType {
   DATE,
   DATETIME,
   BINARY,
-  DECIMAL,
+  DECIMAL_DEPRACTED, // not used now, only for place holder
```
Comment (morningman) [inline gensrc/thrift/Types.thrift:69]: Is it compatible to change the enum's name?

File: be/src/exprs/expr_context.cpp (hunk 1 above)
Comment (yangzhg) [reply be/src/exprs/expr_context.cpp:204]: OK

File: gensrc/thrift/Types.thrift (hunk 3 above)
Comment (yangzhg) [reply gensrc/thrift/Types.thrift:69]: we don't use name literal in our code , just use as int

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**